app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -484,6 +508,14 @@ def construct_repo_from_params(repo_params) # rubocop:disable Metrics/AbcSize
       root.os_versions = repo_params.fetch(:os_versions, []) if root.yum?
       root.retain_package_versions_count = repo_params[:retain_package_versions_count] if root.yum? && repo_params.key?(:retain_package_versions_count)
 
+      if root.generic?
+        generic_remote_options = {}
+        RepositoryTypeManager.generic_remote_options.each do |option|
+          generic_remote_options[option.name] = repo_params[option.name]
+        end
+        root.generic_remote_options = generic_remote_options.to_json
```

Comment:
You'll need to do this kind of update for ::Actions::Katello::Repository::Update too.  Perhaps you can edit the repo_params before they get sent to the actoin in `def update` ?

Reply:
@ianballou i ended up doing it this way. my idea was getting too complicated.